src/main/java/org/openhab/automation/jrule/actions/JRuleActionClassGenerator.java

```diff
@@ -124,6 +125,14 @@ private Map<String, Object> createActionsModel(Thing thing) {
         Map<String, Object> freemarkerModel = new HashMap<>();
         freemarkerModel.put("id", thing.getUID().toString());
         freemarkerModel.put("scope", thing.getUID().getBindingId());
+        if (thing.getHandler() != null) {
+            Class<? extends ThingHandlerService> thingActionsClass = thing.getHandler().getServices().stream()
+                    .filter(ThingActions.class::isAssignableFrom).findFirst()
+                    .orElseThrow(() -> new IllegalStateException("should not occur here"));
+            if (thingActionsClass.getAnnotation(ThingActionsScope.class) != null) {
+                freemarkerModel.put("scope", thingActionsClass.getAnnotation(ThingActionsScope.class).name());
+            }
+        }
```

Comment:
Better with a if/else to set the scope variable?
maybe with Optional.of() to handle the two if statements.